File: CS-Examples/09_Charts/RichTextForDataLabel.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Xls;
using Spire.Xls.Charts;


namespace RichTextForDataLabel
{
	public partial class Form1 : Form
	{
        public Form1()
        {
            InitializeComponent();
        }
		private void btnRun_Click(object sender, System.EventArgs e)
        {
            //Create a workbook
            Workbook workbook = new Workbook();

            //Load the Excel document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ChartToImage.xlsx");

            //Get first worksheet of the workbook
            Worksheet worksheet = workbook.Worksheets[0];

            //Get the first chart inside this worksheet
            Chart chart = worksheet.Charts[0];

            //Get the first datalabel of the first series 
            ChartDataLabels datalabel = chart.Series[0].DataPoints[0].DataLabels;

            //Set the text
            datalabel.Text = "Rich Text Label";

            //Show the value
            chart.Series[0].DataPoints[0].DataLabels.HasValue = true;

            //Set styles for the text
            chart.Series[0].DataPoints[0].DataLabels.Color = Color.Red;
            chart.Series[0].DataPoints[0].DataLabels.IsBold = true;

            //String for output file 
            String outputFile = "Output.xlsx";

            //Save the file
            workbook.SaveToFile(outputFile, ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //Launching the output file.
            Viewer(outputFile);
		}
		private void Viewer( string fileName )
		{
			try
			{
				System.Diagnostics.Process.Start(fileName);
			}
			catch{}
		}
        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

	}
}

File: VB-Examples/09_Charts/RichTextForDataLabel.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel
Imports System.IO
Imports Spire.Xls
Imports Spire.Xls.Charts
Imports System.Text

Namespace RichTextForDataLabel
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Load the Excel file from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ChartToImage.xlsx")

            ' Get the first worksheet from the workbook
            Dim worksheet As Worksheet = workbook.Worksheets(0)

            ' Get the first chart from the worksheet
            Dim chart As Chart = worksheet.Charts(0)

            ' Get the data labels of the first series and first data point in the chart
            Dim datalabel As ChartDataLabels = chart.Series(0).DataPoints(0).DataLabels

            ' Set the text of the data label to "Rich Text Label"
            datalabel.Text = "Rich Text Label"

            ' Enable data labels for the first data point in the first series
            chart.Series(0).DataPoints(0).DataLabels.HasValue = True

            ' Set the font color of the data label to red
            chart.Series(0).DataPoints(0).DataLabels.Color = Color.Red

            ' Set the font style of the data label to bold
            chart.Series(0).DataPoints(0).DataLabels.IsBold = True

            ' Specify the output file name
            Dim outputFile As String = "Output.xlsx"

            ' Save the modified workbook to the specified file with Excel 2013 format
            workbook.SaveToFile(outputFile, ExcelVersion.Version2013)

            ' Release the resources used by the workbook
            workbook.Dispose()

            'Launching the output file.
            Viewer(outputFile)
		End Sub
		Private Sub Viewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub

	End Class
End Namespace
